common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -61,6 +67,13 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](
   private val client: CouchDbRestClient =
     new CouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, dbName)
 
+  // This the the amount of allowed parallel requests for each entity, before batching starts. If there are already maxOpenDbRequests
+  // and more documents need to be stored, then all arriving documents will be put into batches (if enabled) to avoid a long queue.
+  private val maxOpenDbRequests = system.settings.config.getInt("akka.http.host-connection-pool.max-open-requests") / 2
```

Comment:
Why half of `max-open-requests`?

Reply:
To prevent fights of other put/reads vs. activation puts.

Reply:
Due to the current design, we have an datastore (and a batcher) for each entity. That means in the controller we have 5 batchers. All of them are sharing the 128 connections, that are configured in akka (max-open-requests).
If the controller has to write a lot of activations now, only 64 of these connections will be used on setting `maxOpenDbRequests` to half of `max-open-requests`.
So if someone else comes and wants to invoke an action that is not in the cache, it has not to wait until all the activations are written into the database.
In the invoker that also means, that writing activations will not affect getting actions from the database.
In addition, @markusthoemmes and me did some performance meassurements and there was no big difference between using 64 or 128 requests in parallel to write away activations.